src/Game/MeshBuilder.cpp

```diff
@@ -84,13 +84,18 @@ Mesh createMesh(const Wall& wall)
             });
     }
 
-    //Textre Coords (To do)
-    mesh.texCoords = {
-        1.0f, 1.0f, 1.0f, 0.0f, 0.0f, 0.0f, 0.0f, 1.0f
-    };
+    //Textre Coords
+	float length = sqrt(pow(wallFinish.y - wallOrigin.y, 2) + pow(wallFinish.x - wallOrigin.x, 2));
+	float x_2d = length / WORLD_SIZE;
+	const sf::Vector2f tSize = wTex.getTextureScale(CY_COLOR);
```

Comment:
CY_COLOR? What happens if it is not a colour

Reply:
No wonder the texture scaling was acting up, good spot